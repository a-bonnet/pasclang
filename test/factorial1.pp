program
var
    a : integer;
function factorial(n : integer) : integer;
var
    i : integer;
begin
    factorial := 1 ;
    i := 1 ;
    while i <= n do
    begin
        factorial := i * factorial ;
        i := i + 1
    end
end;
begin
    a := 1;
    while (a > 0) do
        begin
            a := readln();
            writeln(factorial(a))
        end
end.
