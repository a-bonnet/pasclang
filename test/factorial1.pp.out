120
1
720
6
1
